### PR TITLE
Update django 4.1.7

### PR DESCRIPTION
**Pull request by norkans7**

### CODE REVIEW

**Comment by rowanseymour** [comment]
@norkans7 please add some comments here explaining _why_ changes are needed - for bonus points could even reference the relevant parts of the release notes

File: temba/channels/types/external/tests.py [hunk 1]
```diff
@@ -79,13 +79,13 @@ def test_claim(self):
 
         # test substitution in our url
         self.assertEqual(
-            "http://test.com/send.php?from=5080&text=test&to=%2B250788383383",
+            "http://example.com/send.php?from=5080&text=test&to=%2B250788383383",
```
Comment (norkans7) [inline temba/channels/types/external/tests.py:82]: test.com cannot be resolved so changed that to example.com 

I think we could add some mock for completely have the tests run offline but that is a different PR

File: temba/classifiers/types/luis/tests.py [hunk 2]
```diff
@@ -217,7 +217,7 @@ def test_connect(self, mock_get_host, mock_predict, mock_get_version_intents, mo
                 "slot": "production",
             },
         )
-        self.assertFormError(response, "form", "__all__", "Check prediction credentials: Not authorized")
```
Comment (norkans7) [inline temba/classifiers/types/luis/tests.py:220]: `To check the form’s non-field errors, use field=None.`

https://docs.djangoproject.com/en/4.1/topics/testing/tools/#django.test.SimpleTestCase.assertFormError

File: temba/contacts/views.py [hunk 3]
```diff
@@ -1592,7 +1592,7 @@ def clean_name(self):
     def clean_value_type(self):
         value_type = self.cleaned_data["value_type"]
 
-        if self.instance and self.instance.campaign_events.filter(is_active=True).exists():
+        if self.instance and self.instance.id and self.instance.campaign_events.filter(is_active=True).exists():
```
Comment (norkans7) [inline temba/contacts/views.py:1595]: Now, the create view have an instance that is not yet saved so we check that is has an ID as a way to differentiate that from update views for an existing instance in the DB

File: temba/msgs/tests.py [hunk 4]
```diff
@@ -350,7 +350,7 @@ def test_delete(self, mock_storage_delete):
         # test a hard delete as part of a contact removal
         msg2.delete()
         self.assertEqual(2, Msg.objects.all().count())
-        self.assertEqual(0, msg2.channel_logs.count())  # logs should be gone
+        self.assertEqual(0, ChannelLog.objects.filter(msg_id=msg2.id).count())  # logs should be gone
```
Comment (norkans7) [inline temba/msgs/tests.py:353]: proper way to test this as related name connections have been cleared by delete

File: temba/contacts/models.py [hunk 5]
```diff
@@ -1231,6 +1231,9 @@ def get_urns(self):
         if hasattr(self, cache_attr):
             return getattr(self, cache_attr)
 
+        if not self.id:
+            return []
```
Comment (rowanseymour) [inline temba/contacts/models.py:1235]: why?

File: temba/contacts/tests.py [hunk 6]
```diff
@@ -2492,10 +2492,10 @@ def test_msg_status_badge(self):
         self.assertIn('"x"', msg_status_badge(msg))
 
     def test_history_templatetags(self):
-        item = {"type": "webhook_called", "url": "http://test.com", "status": "success"}
+        item = {"type": "webhook_called", "url": "http://example.com", "status": "success"}
         self.assertEqual(history_class(item), "non-msg detail-event")
 
-        item = {"type": "webhook_called", "url": "http://test.com", "status": "response_error"}
+        item = {"type": "webhook_called", "url": "http://exmaple.com", "status": "response_error"}
```
Comment (rowanseymour) [inline temba/contacts/tests.py:2498]: exmaple 😆

File: temba/contacts/models.py (hunk 5 above)
Comment (rowanseymour) [reply temba/contacts/models.py:1235]: if this is related to the contact create form having an id-less contact, I think best to keep that logic in the form

Comment (norkans7) [reply temba/contacts/models.py:1235]: This method is called in the create and update views, so the related name will only for the update view when we have an id on the instance

File: temba/channels/types/external/tests.py (hunk 1 above)
Comment (rowanseymour) [reply temba/channels/types/external/tests.py:82]: yeah pleae look into this in a different pr - our unit tests should never be hitting real world external sites

**Review by rowanseymour — APPROVED** [review]

**Pull request merged**